source/resource/blocks.json

```
@@ -7,26 +7,118 @@
           "properties": {
-            "down": { "type": "string", "pattern": "^[\\w_\\-]+$" },
-            "up": { "type": "string", "pattern": "^[\\w_\\-]+$" },
-            "side": { "type": "string", "pattern": "^[\\w_\\-]+$" },
-            "south": { "type": "string", "pattern": "^[\\w_\\-]+$" },
-            "north": { "type": "string", "pattern": "^[\\w_\\-]+$" },
-            "west": { "type": "string", "pattern": "^[\\w_\\-]+$" },
-            "east": { "type": "string", "pattern": "^[\\w_\\-]+$" }
+            "down": {
+              "type": "string",
+              "pattern": "^[\\w_\\-]+$"
+            },
+            "up": {
+              "type": "string",
+              "pattern": "^[\\w_\\-]+$"
+            },
+            "side": {
+              "type": "string",
+              "pattern": "^[\\w_\\-]+$"
+            },
+            "south": {
+              "type": "string",
+              "pattern": "^[\\w_\\-]+$"
+            },
+            "north": {
+              "type": "string",
+              "pattern": "^[\\w_\\-]+$"
+            },
+            "west": {
+              "type": "string",
+              "pattern": "^[\\w_\\-]+$"
+            },
+            "east": {
+              "type": "string",
+              "pattern": "^[\\w_\\-]+$"
+            }
           }
         }
       ]
+    },
+    "A": {
+      "title": "Format Version",
+      "description": "A version that tells minecraft what type of data format can be expected when reading this file.",
+      "pattern": "^([1-9]+)\\.([0-9]+)\\.([0-9]+)$",
+      "type": "string",
+      "default": "1.19.40",
+      "examples": [
+        "1.19.40",
+        "1.18.0",
+        "1.17.0",
+        "1.16.0",
+        "1.15.0",
+        "1.14.0",
+        "1.13.0",
+        "1.12.0",
+        "1.10.0",
+        "1.8.0"
+      ],
+      "defaultSnippets": [
+        {
+          "label": "New Format version",
+          "body": "1.${1|8,10,12,17,18,19|}.${3|2|0|}"
+        }
+      ]
+    },
+    "B": {
+      "title": "Format Version",
+      "description": "A version that tells minecraft what type of data format can be expected when reading this file.",
+      "type": "array",
+      "default": [
+        1,
+        1,
+        0
+      ],
+      "examples": [
+        [
+          1,
+          1,
+          0
+        ]
+      ],
+      "items": {
+        "type": "integer"
+      },
+      "minItems": 3,
+      "maxItems": 3,
+      "uniqueItems": false,
+      "defaultSnippets": [
+        {
+          "label": "New Format version",
+          "body": [
+            1,
+            1,
+            0
+          ]
+        }
+      ]
     }
   },
-  "propertyNames": { "pattern": "^[\\w_\\-:\\.]+$" },
+  "propertyNames": {
+    "pattern": "^[\\w_\\-:\\.]+$"
+  },
   "properties": {
-    "format_version": { "$ref": "../general/format_version.json" }
+    "format_version": {
+      "oneOf": [
+        {
+          "$ref": "#/definitions/A"

```

No need to change this, still and the automated software will take care of it 😄 

```suggestion
          "$ref": "../general/format_version.json"
```